src/serac/physics/functional_residual.hpp

```diff
@@ -121,67 +126,71 @@ class FunctionalResidual<spatial_dim, OutputSpace, Parameters<InputSpaces...>,
   /**
    * @brief Set the Neumann boundary condition
    *
-   * @tparam NeumannType The type of the traction load
    * * // DependsOn<active_parameters...> can be indices into fields which the body integral may depend on
-   * @param boundary_name The name of the registered domain over which the traction is applied. If nothing is supplied
-   * the entire boundary is
-   * @param surface_function A function describing the traction applied to a boundary
-   * used.
+   * @tparam NeumannType The type of the traction load
+   * @param boundary_name The name of the registered domain over which the traction is applied.
+   * @param integrand A function describing the traction applied to a boundary
+   * used.  Our convention for the sign of the residual
+   * vector is that it is expected to be a 'negative force', so the mass terms show up with a positive sign in the
+   * residual.  This also ensures that the Jacobian of the residual is positive definite for most physics.  A body
+   * integrand involving 'right hand side' contributions like a body load, should be supplied by the user with a
+   * negative sign.
    * @pre NeumannType must be a object that can be called with the following arguments:
    *    1. `double t` the time
-   *    1. `tensor<T,dim> x` the spatial coordinates for the quadrature point
+   *    1. `tensor<T,dim> X` the spatial coordinates for the quadrature point
    *    3. `tensor<T,dim> n` the outward-facing unit normal for the quadrature point
    *    4. `tuple{value, derivative}`, a variadic list of tuples (each with a values and derivative),
    *            one tuple for each of the trial spaces specified in the `DependsOn<...>` argument.
-   *
    * @note The actual types of these arguments passed will be `double`, `tensor<double, ... >` or tuples thereof
    *    when doing direct evaluation. When differentiating with respect to one of the inputs, its stored
    *    values will change to `dual` numbers rather than `double`. (e.g. `tensor<double,3>` becomes `tensor<dual<...>,
    * 3>`)
    *
    */
   template <int... active_parameters, typename NeumannType>
-  void addBoundaryIntegral(DependsOn<active_parameters...>, std::string boundary_name, NeumannType surface_function)
+  void addBoundaryIntegral(DependsOn<active_parameters...>, std::string boundary_name, NeumannType integrand)
   {
     residual_->AddBoundaryIntegral(
         Dimension<spatial_dim - 1>{}, DependsOn<active_parameters...>{},
-        [surface_function](double t, auto X, auto... params) {
+        [integrand](double t, auto X, auto... params) {
           auto n = cross(get<DERIVATIVE>(X));
-          return surface_function(t, get<VALUE>(X), normalize(n), params...);
+          return integrand(t, get<VALUE>(X), normalize(n), params...);
```

Comment:
Should we sent in `X` with its derivatives to the integrand (not just the value) so that users can compute normals in the current configuration?